pkg/apis/serving/v1alpha2/kfservice_defaults.go

```diff
@@ -19,8 +19,8 @@ package v1alpha2
 // Default implements https://godoc.org/sigs.k8s.io/controller-runtime/pkg/webhook/admission#Defaulter
 func (kfsvc *KFService) Default() {
 	logger.Info("Defaulting KFService", "namespace", kfsvc.Namespace, "name", kfsvc.Name)
-	kfsvc.Spec.Default.ApplyDefaults()
+	kfsvc.Spec.Default.Predict.ApplyDefaults()
```

Comment:
Leave this as-is and implement ComponentSpec.ApplyDefaults instead?

Reply:
I am trying to limit the scope of this PR to v1alpha2 spec change, other changes are really just to  fix the compilation, I am the sure the follow up PR will change this once we extend to explain and transform.